src/ekstat.erl

```diff
@@ -20,38 +26,76 @@
 
 -spec open() -> {ok, Handle::kstat_handle()} | kstat_error().
 open() ->
-    erlang:nif_error(undefined).
+    ?ASYNC_NIF_CALL(fun open_nif/1, []).
+
+-spec close(Handle::binary()) -> ok | kstat_error().
+close(Handle) ->
+    ?ASYNC_NIF_CALL(fun close_nif/2, [Handle]).
+
+-spec clear(Handle::binary()) -> {ok, NumInstances::integer()} | kstat_error().
+clear(Handle) ->
+    ?ASYNC_NIF_CALL(fun clear_nif/2, [Handle]).
 
 -spec update(Handle::binary()) -> {ok, NumInstances::integer()} | kstat_error().
-update(_Handle) ->
+update(Handle) ->
+    ?ASYNC_NIF_CALL(fun update_nif/2, [Handle]).
+
+-spec read(Handle::binary()) -> [kstat_row()] | kstat_error().
+read(Handle) ->
+    ?ASYNC_NIF_CALL(fun read_nif/2, [Handle]).
+
+-spec read(Handle::binary(), Class::kstat_pattern()) -> [kstat_row()] | kstat_error().
+read(Handle, Class) ->
+    ?ASYNC_NIF_CALL(fun read_nif/3, [Handle, Class]).
+
+-spec read(Handle::binary(), Class::kstat_pattern(), Module::kstat_pattern()) -> [kstat_row()] | kstat_error().
+read(Handle, Class, Module) ->
+    ?ASYNC_NIF_CALL(fun read_nif/4, [Handle, Class, Module]).
+
+-spec read(Handle::binary(), Class::kstat_pattern(), Module::kstat_pattern(), Instance::kstat_pattern()) -> [kstat_row()] | kstat_error().
+read(Handle, Class, Module, Instance) ->
+    ?ASYNC_NIF_CALL(fun read_nif/5, [Handle, Class, Module, Instance]).
+
+-spec read(Handle::binary(), Class::kstat_pattern(), Module::kstat_pattern(), Instance::kstat_pattern(), Name::kstat_pattern()) -> [kstat_row()] | kstat_error().
```

Comment:
Line 59 is too long: -spec read(Handle::binary(), Class::kstat_pattern(), Module::kstat_pattern(), Instance::kstat_pattern(), Name::kstat_pattern()) -> [kstat_row()] | kstat_error()..